File: src/jump-game/jump-game.ts
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canJump = function(nums) {
    let finalPos = nums.length - 1;
    for (let i = nums.length -1; i > -1; i-=1) {
        if (nums[i] + i >= finalPos) {
            finalPos = i
        }
    }
    return finalPos === 0;
};

File: src/jump-game/jump-game.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        n = len(nums)
        final_pos =  -1
        for i in range(n-1, -1, -1):
            if (i + nums[i] >= final_pos):
                final_pos = i
        return True if final_pos == 0 else False
        